ts/src/test/scala/ch/epfl/ts/test/data/StrategyParameters.scala

```diff
@@ -0,0 +1,105 @@
+
+@RunWith(classOf[JUnitRunner])
+class StrategyParametersTests extends FunSuite {
+  /**
+   * Common test values
+   */
+	val currencies1 = (Currency.EUR, Currency.CHF)
+	val currencies2 = (Currency.GBP, Currency.USD)
+  val coefficient1 = 0.2342341341
+  val coefficient2 = -0.2342341341
+  
+  private type CurrencyPair = (Currency.Currency, Currency.Currency)
+  
+  /** Generic function to test basic functionality expected from Parameter subclasses */
+  def testConcreteParameter[T](parameterTrait: ParameterTrait[T], validValues: Iterable[T], invalidValues: Iterable[T]) = {
+    test(parameterTrait + " should give back the value it was instantiated with") {
+      for {
+        myValue <- validValues
+        parameter = parameterTrait.getInstance(myValue)        
+      } yield assert(parameter.get() == myValue, "Should give back value " + myValue)
+    }
+    
+    test(parameterTrait + " should reject invalid values") {
+      for {
+        myValue <- invalidValues
+        attempt = Try(parameterTrait.getInstance(myValue))
+      } yield assert(attempt.isFailure, "Should fail with illegal value " + myValue)
+    }
+  
+    test(parameterTrait + " should accept all values within the acceptable range") {
+      for {
+        value <- parameterTrait.validValues
+        attempt = Try(parameterTrait.getInstance(value))
+      } yield assert(attempt.isSuccess, "Should accept " + value)
+    }
+    
+    test(parameterTrait + " should have a default value which is valid") {
+    	val attempt = Try(parameterTrait.getInstance(parameterTrait.defaultValue))
+    	assert(attempt.isSuccess, "Should accept " + parameterTrait.defaultValue)
+    }
+  }
+  
+  test("Should allow to add several parameters at a time") {
+    val myParameters = new StrategyParameters(
+      "tradedCurrencies" -> CurrencyPairParameter(currencies1),
+      "someCoefficient" -> CoefficientParameter(coefficient1),
+      "someOtherParameter" -> CurrencyPairParameter(currencies2)
+    )
+  }
+  
+  test("Should fail at instantiation with illegal parameters") {
+	  val attempt = Try(new StrategyParameters("someCoefficient" -> CoefficientParameter(coefficient2)))
+    assert(attempt.isFailure, "Should fail to instantiate a coefficient with value " + coefficient2)
+  }
+  
+  test("Should hold the parameters and yield back their values") {
+	  val myParameters = new StrategyParameters(
+      "tradedCurrencies" -> CurrencyPairParameter(currencies1),
+      "someCoefficient" -> CoefficientParameter(coefficient1)
+    )
+    
+    assert(myParameters.get[CurrencyPair]("tradedCurrencies") == Some(currencies1))
+    assert(myParameters.get[Double]("someCoefficient") == Some(coefficient1))
+  }
+  
+  test("Should not yield a value if it doesn't have the expected type") {
+    val myParameters = new StrategyParameters("tradedCurrencies" -> CoefficientParameter(coefficient1))
+    
+    val got = myParameters.get[CurrencyPair]("tradedCurrencies")
```

Comment:
You can also get the example `myParameters.get[CoefficientParameter]("unknown_key")` - it should behave the same, as far as I understand.